edx_exams/apps/api/serializers.py

```diff
@@ -40,6 +47,73 @@ class Meta:
         fields = ['name', 'verbose_name', 'lti_configuration_id']
 
 
+class CourseExamConfigurationWriteSerializer(serializers.ModelSerializer):
+    """
+    Serializer for writing to the CourseExamConfiguration model.
+
+    We have separate read and write serializers because the read serializer uses the name field of the provider instance
```

Comment:
Based on an earlier conversation with Zach, I decided it would be better to have two separate serializers than to have to worry about someone reading the data from the serializer improperly. This will be less error prone, and it's explicit that there are two different serializers - one for reading and one for writing. I can revert this to the old code (see my first commit), but I think this is an improvement.